packages/design-system/src/organisms/formStepHeader/index.tsx

```diff
@@ -0,0 +1,125 @@
+import { useEffect, useMemo, useRef } from 'react'
+import { FlatList, Platform, StyleSheet, View } from 'react-native'
+import { TypographyV2 } from '../../atoms/typographyV2'
+import { useThemeV2 } from '../../hooks'
+import { container, margin, spacing } from '../../styles'
+import { makeScrollToIndexFailedHandler } from '../../utils'
+
+const STEP_INDEX_SIZE = spacing[24]
+const STEP_INDEX_BORDER_RADIUS = 24
+const STEP_INDEX_BORDER_WIDTH = 2
+const SEPARATOR_WIDTH = 48
+const SEPARATOR_HEIGHT = 2
+const NATIVE_LINE_HEIGHT = 20
+
+const styles = StyleSheet.create({
+  stepIndex: {
+    width: STEP_INDEX_SIZE,
+    height: STEP_INDEX_SIZE,
+    borderWidth: STEP_INDEX_BORDER_WIDTH,
+    borderRadius: STEP_INDEX_BORDER_RADIUS,
+  },
+  separator: {
+    width: SEPARATOR_WIDTH,
+    height: SEPARATOR_HEIGHT,
+  },
+})
+
+const Separator: React.FC = () => {
+  const theme = useThemeV2()
+  return (
+    <View style={[container.fullHeight, container.center]}>
+      <View
+        style={[
+          margin.mh16,
+          styles.separator,
+          { backgroundColor: theme.background.silver },
+        ]}
+      />
+    </View>
+  )
+}
+
+interface StepProps {
+  step: number
+  title: string
+  isChecked: boolean
```

Comment:
Let's follow the same naming convention as `disabled` & `editable` here:

```suggestion
  checked: boolean
```